verifiers/utils/dsse.go

```diff
@@ -1,13 +1,36 @@
 package utils
 
 import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/x509"
 	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
 	"fmt"
+	"math/big"
 
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
 	dsselib "github.com/secure-systems-lab/go-securesystemslib/dsse"
 	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
 )
 
+func EnvelopeFromBytes(payload []byte) (*dsselib.Envelope, error) {
+	var env dsselib.Envelope
+	err := json.Unmarshal(payload, &env)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", serrors.ErrorInvalidDssePayload, err)
```

Comment:
nit: wrap both errors
```suggestion
		return nil, fmt.Errorf("%w: %w", serrors.ErrorInvalidDssePayload, err)
```